src/ReactLeafletMap/layers/SvgVectorLayer/Feature/MarkerShape.js

```diff
@@ -0,0 +1,184 @@
+		// Basic shape -> no need for svg
+		if (this.basicShape) {
+			const style = helpers.getMarkerShapeCssStyle(this.style);
+			return ReactDOMServer.renderToString(<div style={style}/>);
+		}
+
+		else {
+			let finalShape;
+			const style = helpers.getMarkerShapeSvgStyle(this.style);
+
+			// Combined shape and icon
+			// Currently only shape="pin" is suitable for combination
+			if (this.shape && this.icon) {
+				finalShape = this.getShapeWithIcon(style);
+			}
+
+			// Just shape
+			else if (this.shape) {
+				finalShape = this.getShape(style);
+			}
+
+			// No shape, but icon? Use icon as shape
+			else if (this.icon) {
+				finalShape = this.getIcon(style);
+			}
+
+			return ReactDOMServer.renderToString(finalShape);
+		}
+	}
+
+	/**
+	 * @param style {Object} style object suitable for SVG
+	 * @return {React.DetailedReactHTMLElement<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>}
+	 */
+	getShape(style) {
+		const props = this.shape.componentProps ? {...this.shape.componentProps, style} : {style};
+		return React.createElement(this.shape.component, {...props, outlineWidth: this.style.weight});
+	}
+
+	/**
+	 * @param style {Object} style object suitable for SVG
+	 * @return {React.DetailedReactHTMLElement<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>}
+	 */
+	getIcon(style) {
+		const iconStyle = {
+			...style,
+			fill: style.iconFill ? style.iconFill : style.fill
+		}
+
+		const props = this.icon.componentProps ? {...this.icon.componentProps, style: iconStyle} : {style: iconStyle};
+		return React.createElement(this.icon.component, props);
+	}
+
+	/**
+	 * @param style {Object} style object suitable for SVG
+	 * @return {React.DetailedReactHTMLElement<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>}
+	 */
+	getShapeWithIcon(style) {
+		let iconStyle = {strokeWidth: 0}; // TODO think about icons styling inside shape
+		const {iconFill, ...shapeStyle} = style;
+
+		if (iconFill) {
+			iconStyle.fill = iconFill;
+		} else {
+			iconStyle.fill = shapeStyle.stroke;
+		}
+
+		const iconProps = this.icon.componentProps ? {...this.icon.componentProps, style: iconStyle} : {style: iconStyle};
+		const iconComponent = React.createElement(this.icon.component, iconProps);
+
+		const props = this.shape.componentProps ? {...this.shape.componentProps, style: shapeStyle} : {style: shapeStyle};
+		return React.createElement(this.shape.component, {...props, icon: iconComponent, outlineWidth: this.style.weight});
+	}
+
+	/**
+	 * Set style of element
+	 * @param style {Object} Leaflet style definition
+	 * @param id {string} id of the shape
+	 * @param isBasicShape {bool} If true -> DIV element, if false -> SVG element
+	 */
+	setStyle(style, id, isBasicShape) {
+		id = id || this.id;
+		isBasicShape = isBasicShape || this.basicShape;
+
+		let shapeStyle = isBasicShape ? helpers.getMarkerShapeCssStyle(style) : helpers.getMarkerShapeSvgStyle(style);
+		let element = document.getElementById(id);
```

Comment:
Can you replace "document" with ReactDOM? Otherwise there will bo conflict on SSR. Same on line 55

Reply:
I can't find an easy solution for this. Let's postpone it for now.